a.go

```diff
@@ -320,10 +332,10 @@ func (a *A) runCleanups(finished *bool, panicVal *interface{}, panicStack *[]byt
 	for {
 		var cleanup func()
 		a.mu.Lock()
-		if len(a.cleanups) > 0 {
-			last := len(a.cleanups) - 1
-			cleanup = a.cleanups[last]
-			a.cleanups = a.cleanups[:last]
+		if len(*a.cleanups) > 0 {
+			last := len(*a.cleanups) - 1
+			cleanup = (*a.cleanups)[last]
+			*a.cleanups = (*a.cleanups)[:last]
```

Comment:
I think it is not safe - because the mutex is captured on a copy (one of them) and this will lead to a race.

Reply:
For sure it would be good to add some concurrent tests

Reply:
```
$ go test -race -count=100
PASS
ok      github.com/goyek/goyek/v2       1.919s
```

Reply:
Tests added https://github.com/goyek/goyek/pull/460/commits/1c724f71a5f147e58178a311340e132181c6f053

```
$ go test -race -count=1000
PASS
ok      github.com/goyek/goyek/v2       6.966s
```